### ISSUE TITLE
EnvSettingsSource get_field_value returns field_name instead of the env name.

### ISSUE THREAD

**Issue by atti92** [pydantic/pydantic-settings#504]
In 2.7.0 get_field_value returns the field_name instead of the env name, we depended on this logic, but now it makes it really hard to figure out which environment variable was used.

Can we have the env_name added back to EnvSettingsSource to look up the value?

I can see why the change was made, and it makes sense, but we are loosing data with it. We were using it to give feedback for users to track each configuration's value's source. eg.:   conf.value1 -> $MY_ENVVAR,  conf.value2 -> .env/VALUE2, etc...

Code block in question:
```
        for field_key, env_name, value_is_complex in field_infos:
            env_val = self.env_vars.get(env_name)
            if env_val is not None:
                if value_is_complex or (self.config.get('populate_by_name', False) and (field_key == field_name)):
                    preferred_key = field_key
                break

        return env_val, preferred_key, value_is_complex
```
 We are also fine if env_name is saved somewhere inside the class to lookup later.

**Comment by hramezani** [comment]
Thanks @kschwab for the PR.

The PR will introduce a breaking change because it changes the signature of `get_field_value`.

**Comment by kschwab** [comment]
Sure @hramezani. Would you like me to create a new public API for environment sources, e.g., `get_resolved_field_value`, that returns the preferred alias key, or just keep the resolution private?

**Comment by hramezani** [comment]
let's not introduce a new API for now and keep it private.

But it would be good to add a comment for the future. we might change it in V3

**Comment by kschwab** [comment]
I restored the original `get_field_value` behavior and moved the alias resolution into a private method. If desired it can be made public in V3.

**Comment by hramezani** [comment]
The fix has been released on [pydantic-settings 2.7.1](https://pypi.org/project/pydantic-settings/)
